Source/FakeItEasy.Specs/DisposableSpecs.cs

```diff
@@ -0,0 +1,55 @@
+﻿namespace FakeItEasy.Specs
+{
+    using System;
+    using FluentAssertions;
+    using Machine.Specifications;
+
+    public class when_faking_a_disposable_class
+        : EventRaisingSpecs
+    {
+        static IDisposable fake;
+        private static Exception exception;
+
+        Establish context = () =>
+        {
+            AppDomain.CurrentDomain.UnhandledException += UnhandledExceptionHandler;
+
+            fake = A.Fake<SomeDisposable>();
+        };
+
+        Because of = () =>
+        {
+            fake = null;
+            GC.Collect();
+            GC.WaitForPendingFinalizers();
+        };

```

Comment:
And I thought I was the champion of "conversations with myself in fora and PRs".

Yup, it failed. Your way looked good too, but since it was more complicated and, as you say, these sorts of tests can never _prove_ anything, I figured, "go for the low line count".
I am in _no way_ offended that you tried it yourself. Prudent.

To tell the truth, I didn't understand why you had so many collects. What was the benefit of checking that "the deallocation of everything else happens before the finalization of the fake"? The fake's interceptor holds the `FakeManager`, so everything that _mattered_ would all still be live until we break reachability of the fake.

And thanks for the speedy review and merge!